var/spack/repos/builtin/packages/xsdk/package.py

```diff
@@ -18,91 +19,128 @@ class Xsdk(BundlePackage):
-    depends_on('petsc@xsdk-0.2.0+trilinos+mpi+hypre+superlu-dist+metis+hdf5~mumps+double~int64',
+    depends_on('petsc@xsdk-0.2.0+mpi+hypre+superlu-dist+metis+hdf5~mumps+double~int64',
                when='@xsdk-0.2.0')
 
-    depends_on('dealii@develop~assimp~python~doc~gmsh+petsc+slepc+mpi+trilinos~int64+hdf5~netcdf+metis~sundials~ginkgo~symengine', when='@develop +dealii')
-    depends_on('dealii@9.0.1~assimp~python~doc~gmsh+petsc~slepc+mpi+trilinos~int64+hdf5~netcdf+metis~ginkgo~symengine', when='@0.4.0 +dealii')
+    depends_on('dealii +trilinos', when='+trilinos +dealii')
+    depends_on('dealii ~trilinos', when='~trilinos +dealii')
+    depends_on('dealii@develop~assimp~python~doc~gmsh+petsc+slepc+mpi~int64+hdf5~netcdf+metis~sundials~ginkgo~symengine', when='@develop +dealii')
+    depends_on('dealii@9.1.1~assimp~python~doc~gmsh+petsc+slepc+mpi~int64+hdf5~netcdf+metis~sundials~ginkgo~symengine', when='@0.5.0 +dealii')
+    depends_on('dealii@9.0.1~assimp~python~doc~gmsh+petsc~slepc+mpi~int64+hdf5~netcdf+metis~ginkgo~symengine', when='@0.4.0 +dealii')
 
     depends_on('pflotran@develop', when='@develop')
+    depends_on('pflotran@xsdk-0.5.0', when='@0.5.0')
     depends_on('pflotran@xsdk-0.4.0', when='@0.4.0')
     depends_on('pflotran@xsdk-0.3.0', when='@0.3.0')
     depends_on('pflotran@xsdk-0.2.0', when='@xsdk-0.2.0')
 
     depends_on('alquimia@develop', when='@develop')
+    depends_on('alquimia@xsdk-0.5.0', when='@0.5.0')
     depends_on('alquimia@xsdk-0.4.0', when='@0.4.0')
     depends_on('alquimia@xsdk-0.3.0', when='@0.3.0')
     depends_on('alquimia@xsdk-0.2.0', when='@xsdk-0.2.0')
 
-    depends_on('sundials@4.1.0~int64+hypre', when='@develop')
+    depends_on('sundials+superlu-dist', when='@0.5.0: %gcc@6.1:')
+    depends_on('sundials@develop~int64+hypre+petsc', when='@develop')
+    depends_on('sundials@5.0.0~int64+hypre+petsc', when='@0.5.0')
     depends_on('sundials@3.2.1~int64+hypre', when='@0.4.0')
     depends_on('sundials@3.1.0~int64+hypre', when='@0.3.0')
 
-    depends_on('plasma@18.11.1:', when='@develop %gcc@6.0:')
+    depends_on('plasma@19.8.1:', when='@develop %gcc@6.0:')
+    depends_on('plasma@19.8.1:', when='@0.5.0 %gcc@6.0:')
     depends_on('plasma@18.11.1:', when='@0.4.0 %gcc@6.0:')
 
-    depends_on('magma@2.4.0', when='@develop +cuda')
+    depends_on('magma@2.5.1', when='@develop +cuda')
+    depends_on('magma@2.5.1', when='@0.5.0 +cuda')
     depends_on('magma@2.4.0', when='@0.4.0 +cuda')
     depends_on('magma@2.2.0', when='@0.3.0 +cuda')
 
-    depends_on('amrex@develop+sundials', when='@develop %intel')
-    depends_on('amrex@develop+sundials', when='@develop %gcc')
+    depends_on('amrex@develop', when='@develop %intel')
+    depends_on('amrex@develop', when='@develop %gcc')
+    depends_on('amrex@19.08', when='@0.5.0 %intel')
+    depends_on('amrex@19.08', when='@0.5.0 %gcc')
     depends_on('amrex@18.10.1', when='@0.4.0 %intel')
     depends_on('amrex@18.10.1', when='@0.4.0 %gcc')
```

Comment:
What exactly does this mean? Amrex isn't needed when building with other compilers?

Reply:
Here I'd like to exclude clang - but couldn't figure out spack syntax for it. [there is a conflict on clang in amrex].

Reply:
Do you want to prevent people from building `xsdk` with clang, or just `amrex`?

Reply:
Most packages build fine with clang. I  want to exclude only `amrex` as it won't build with clang.